app/channels.go

```diff
@@ -65,17 +77,40 @@ type Channels struct {
 }
 
 func init() {
-	RegisterProduct("channels", func(s *Server) (Product, error) {
-		return NewChannels(s)
+	RegisterProduct("channels", func(s *Server, services map[ServiceKey]interface{}) (Product, error) {
+		return NewChannels(s, services)
 	})
 }
 
-func NewChannels(s *Server) (*Channels, error) {
+func NewChannels(s *Server, services map[ServiceKey]interface{}) (*Channels, error) {
```

Comment:
Not urgent but maybe we can define an interface for services eg. `GetName()` to match with the keys.

Reply:
I like it :+1: